src/lib/components/barchartv2/utils.ts

```diff
@@ -36,6 +37,35 @@ export const getMaxValue = (data: { [key: string]: string | number }[]) => {
   });
   return Math.max(...values);
 };
+
+/**
+ * Generates all days between start and end timestamps (inclusive)
+ * @param startTimestamp - Start timestamp in milliseconds
+ * @param endTimestamp - End timestamp in milliseconds
+ * @returns Array of timestamps for each day at midnight
+ */
+const generateDayTimestamps = (
+  startTimestamp: number,
+  endTimestamp: number,
+): number[] => {
+  const dayTimestamps: number[] = [];
+  const startDate = new Date(startTimestamp);
+  const endDate = new Date(endTimestamp);
+
+  // Set to midnight to ensure consistent day boundaries
+  startDate.setHours(0, 0, 0, 0);
+  endDate.setHours(0, 0, 0, 0);
+
+  const currentDate = new Date(startDate);
```

Comment:
Nit: just currentDate is bit confusing for me.
```suggestion
  const iteratingDate = new Date(startDate);
```